### DISCUSSION TITLE
Change player respawn via event and treat the class-defined respawn position as the default

### DISCUSSION THREAD

**Discussion by Xyseer764** [AscensionGameDev/Intersect-Engine#1517] [Ideas]
**Is your feature request related to a problem? Please describe.**
I have no idea why the respawn position is hard-coded into the class. Not being able to change a player's respawn position dynamically heavily restricts what the engine should be able to do. Most major MMOs have some mechanic that allows the player to bind their respawn to a city or home. The majority of games made in this engine would benefit from this feature. I'm aware that you're able to fake this functionality through common events by warping the player on respawn, but this seems like something that should be directly modifiable.

**Describe the solution you'd like**
Keep the class respawn as a default that's set when the character's first created. From there on out allow us to modify the player's respawn point through an Event Command.

**Comment by Cheshire92** [comment]
I've had an idea for this when I implement some other things eventually where you can set it through the event system or default back to the class spawn.

**Comment by jcsnider** [comment]
The general solution to this for now is to have a blank map with an autorun event that checks for conditions and warps accordingly.

If no variables are set it warps to intro maps.

Otherwise it checks variables/classes/other values and can warp accordingly.
